keps/sig-cli/3659-kubectl-apply-prune/README.md

```diff
@@ -627,128 +544,171 @@ In future, we may define additional “index” mechanisms here to further optim
+NAME             TYPE     NAMESPACE   AGE
+myset            Secret   ns1         1d
+
+# Subcommand to list objects in applyset, including children
+$ kubectl apply view-applyset myset -n ns1 --include-members
+NAME                TYPE         NAMESPACE   AGE
+myset               Secret       ns1         1d
+  ∟ myapp           Deployment   ns1         1d
+  ∟ myapp-svc       Service      ns1         1d
+  ∟ myapp-config    ConfigMap    ns1         1d
+
+# Subcommand to delete an applyset and its members
+$ kubectl apply delete-applyset myset -n ns1 --cascade=foreground
+Are you sure you want to delete applyset "myset" in namespace "ns1" and all its members? This will delete the following objects:
+    - Secret/myset
+    - Deployment/myapp
+    - Service/myapp-svc
+    - ConfigMap/myapp-config
+This action cannot be undone. (y/n): y
+Deleting applyset "myset" in namespace "ns1" and all its members...
+Secret/myset deleted
+Deployment/myapp deleted
+Service/myapp-svc deleted
+ConfigMap/myapp-config deleted
+Secret/myset deleted
+
+# Subcommand to migrate from the allowlist-based pruning mode to the new one
+$ kubectl apply generate-applyset myset -n ns1 --selector=app=myapp --legacy-allowlist=apps/v1/Deployment,core/v1/Service,core/v1/ConfigMap
+ApplySet parent Secret "myset" created in namespace "ns1"
+Found and added 3 members to applyset "myset" in namespace "ns1":
+    - Deployment/myapp
+    - Service/myapp-svc
+    - ConfigMap/myapp-config
 ```
 
 
-We intend to treat the flag and any subcommands as alpha commands initially.  During alpha, users will need to set an environment variable (e.g. KUBECTL_APPLYSET) to make the flag available.
+### Kubectl implementation details
+
+Commands will verify that the value of `applyset.kubernetes.io/tooling` has the `kubectl/` prefix before making any mutation, failing with an error if the annotation is present with any other value. It will set this label to `kubectl/vX.XX` (e.g. kubectl/v1.27) when creating/adopting resources as parent objects and update the semver as needed. At least initially, a missing tooling label or blank label value will also be considered an error, though this is not strictly required by the proposed spec and could be relaxed in the future.
 
-Commands will verify that the value of `applyset.k8s.io/tooling` has the `kubectl/` prefix before making any mutation, failing with an error if the annotation is present with any other value. It will set this label to `kubectl/vX.XX` (e.g. kubectl/v1.27) when creating/adopting resources as parent objects and update the semver as needed. At least initially, a missing tooling label or blank label value will also be considered an error, though this is not strictly required by the proposed spec and could be relaxed in the future. We may implement a `--force` flag, but this would likely be logically equivalent in outcome to a full ApplySet deletion and recreation, though with the potential (but not the guarantee) to be less disruptive.
```

Comment:
I don't think we should consider a `--force` flag. For one thing, that flag already exists and means something else. For another, there's already an unwieldy number of flags on apply, so adding another that applies only to an edge case for applyset-based pruning isn't great. Finally, I think this sort of thing could be covered by the repair subcommand?